gen3-load-tests/sdk_testing/download_performance_test.py

```diff
@@ -0,0 +1,2201 @@
+            run_number=run_number,
+            workers=config.max_concurrent_requests_async
+            if "async" in tool_name.lower()
+            else config.num_workers_cdis,
+            total_files=len(manifest_data),
+            successful_downloads=0,
+            success_rate=0,
+            total_download_time=0,
+            total_size_mb=0,
+            average_throughput_mbps=0,
+            files_per_second=0,
+            peak_memory_mb=0,
+            avg_memory_mb=0,
+            peak_cpu_percent=0,
+            avg_cpu_percent=0,
+            setup_time=setup_time,
+            download_time=0,
+            verification_time=0,
+            return_code=-1,
+            error_details=[str(e)],
+        )
+
+
+class CDISDataClientTester:
+    """Enhanced CDIS Data Client tester with profiling and monitoring."""
+
+    def __init__(self, config: TestConfiguration):
+        self.config = config
+        self.timer = PerformanceTimer()
+
+    def test_download_multiple(
+        self, manifest_path: str, run_number: int = 1
+    ) -> PerformanceMetrics:
+        """Test CDIS data client download-multiple functionality with enhanced monitoring."""
+        download_dir = os.path.join(
+            self.config.results_dir, f"cdis_run_{int(time.time())}"
+        )
+
+        cmd = [
+            self.config.gen3_client_path,
+            "download-multiple",
+            "--manifest",
+            manifest_path,
+            "--numparallel",
+            str(self.config.num_workers_cdis),
+            "--download-path",
+            download_dir,
+            "--no-prompt",
+            "--profile",
+            "midrc",
+        ]
+
+        logger = logging.getLogger(__name__)
+        return run_tool_with_profiling(
+            cmd,
+            download_dir,
+            manifest_path,
+            "CDIS Data Client",
+            self.config,
+            run_number,
+            logger,
+        )
+
+
+class Gen3SDKTester:
+    """Enhanced Gen3 SDK tester with async and sync support, profiling and monitoring."""
+
+    def __init__(self, config: TestConfiguration):
+        self.config = config
+        self.timer = PerformanceTimer()
+
+    async def test_download_multiple_async(
+        self, manifest_path: str, run_number: int = 1
+    ) -> PerformanceMetrics:
+        """Test Gen3 SDK download-multiple-async functionality with enhanced monitoring."""
+        if not GEN3_SDK_AVAILABLE:
+            print("  ⚠️  Gen3 SDK not available - skipping async test")
+            return self._create_error_metrics(
+                "Gen3 SDK Async",
+                self.config.max_concurrent_requests_async,
+                "Gen3 SDK not available - check import errors above",
+            )
+
+        download_dir = os.path.join(
+            self.config.results_dir, f"gen3sdk_async_run_{int(time.time())}"
+        )
+
+        cmd = [
+            "python",
+            "-m",
+            "gen3.cli",
```

Comment:
you need to change this, do:

`poetry run gen3` and replace `python -m gen3.cli` b/c otherwise you are potentially getting the Global gen3 installation instead of the one in this virtualenv

Reply:
and once you fix the issues in the SDK branch I commented on, you'll need to update this command:

```
        cmd = [
            "poetry",
            "run",
            "gen3",
            "--auth",
            os.path.abspath(self.config.credentials_path),
            "--endpoint",
            self.config.endpoint,
            "download-multiple",  # NOTE: name change here
            "--manifest",
            os.path.abspath(manifest_path),
            "--download-path",
            os.path.abspath(download_dir),
            "--max-concurrent-requests",
            str(self.config.max_concurrent_requests_async),
            "--filename-format",
            "guid",
            "--no-prompt",
        ]
```